src/ReactiveDomain.Testing/Messaging/ConcurrentMessageQueue.cs

```diff
@@ -59,7 +58,7 @@ public void AssertEmpty()
         {
             if (!IsEmpty)
             {
-                var msg = this.ElementAt(0);
+                TryPeek(out var msg);
```

Comment:
If `TryPeek` fails, we need to still have an appropriate message. If it fails, `msg` will have an unspecified value, so `msg.GetType()` could throw. This needs more robust error handling.

Reply:
Queue is not empty.  We just checked it above.
it is a queue of <Msg> all are messages.  there will be a message object that has a type.
How can you break it? @joshkempner

Reply:
I'd be happy to return the ElementAt(0).

Reply:
Done.